libtenzir/builtins/operators/timeshift.cpp

```diff
@@ -127,12 +128,34 @@ class timeshift_operator final : public crtp_operator<timeshift_operator> {
   std::optional<time> start_ = {};
 };
 
-class plugin final : public virtual operator_plugin<timeshift_operator> {
+class plugin final : public virtual operator_plugin<timeshift_operator>,
+                     public virtual operator_factory_plugin {
 public:
   auto signature() const -> operator_signature override {
     return {.transformation = true};
   }
 
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    auto speed = std::optional<located<double>>{};
+    auto start = std::optional<time>{};
+    auto field = std::string{};
```

Comment:
This should also be an expression/selector. Because `timeshift` changes the underlying value, I think this is a bit harder to implement. You could leave this to a follow-up PR.